### PR TITLE
[goal_distance_calculator]add readme

### PR DESCRIPTION
**Pull request by tkimura4**
## Description

add readme of goal_distance_calculator node

## Review Procedure
Check the readme

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: common/util/goal_distance_calculator/Readme.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+| `deviation/yaw`          | `autoware_debug_msgs::msg::Float64Stamped` | publish yaw deviation of self-pose from goal pose[rad]        |
+| `deviation/yaw_deg`      | `autoware_debug_msgs::msg::Float64Stamped` | publish yaw deviation of self-pose from goal pose[deg]        |
+
+## Parameters
+
+### Node Parameters
+
+| Name          | Type   | Default Value | Explanation            |
+| ------------- | ------ | ------------- | ---------------------- |
+| `update_rate` | double | 10.0          | Timer callback period. |
```
Comment (taikitanaka3) [inline common/util/goal_distance_calculator/Readme.md:33]:
```suggestion
| `update_rate` | double | 10.0          | Timer callback period. [Hz] |
```
format doc would be necessary

**Comment by taikitanaka3** [comment]
@tkimura4 
for some reason building is falied
image

**Review by taikitanaka3 — APPROVED** [review]
LGTM

File: common/util/goal_distance_calculator/Readme.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+| `/tf`                              | `tf2_msgs/TFMessage`                      | TF (self-pose)        |
+
+### Output
+
+| Name                     | Type                                       | Description                                                   |
+| ------------------------ | ------------------------------------------ | ------------------------------------------------------------- |
+| `deviation/lateral`      | `autoware_debug_msgs::msg::Float64Stamped` | publish lateral deviation of self-pose from goal pose[m]      |
+| `deviation/longitudinal` | `autoware_debug_msgs::msg::Float64Stamped` | publish longitudinal deviation of self-pose from goal pose[m] |
+| `deviation/yaw`          | `autoware_debug_msgs::msg::Float64Stamped` | publish yaw deviation of self-pose from goal pose[rad]        |
+| `deviation/yaw_deg`      | `autoware_debug_msgs::msg::Float64Stamped` | publish yaw deviation of self-pose from goal pose[deg]        |
```
Comment (kmiya) [inline common/util/goal_distance_calculator/Readme.md:25]:
```suggestion
| Name                     | Type                                       | Description                                                    |
| ------------------------ | ------------------------------------------ | -------------------------------------------------------------- |
| `deviation/lateral`      | `autoware_debug_msgs::msg::Float64Stamped` | Publish lateral deviation of self-pose from goal pose [m]      |
| `deviation/longitudinal` | `autoware_debug_msgs::msg::Float64Stamped` | Publish longitudinal deviation of self-pose from goal pose [m] |
| `deviation/yaw`          | `autoware_debug_msgs::msg::Float64Stamped` | Publish yaw deviation of self-pose from goal pose [rad]        |
| `deviation/yaw_deg`      | `autoware_debug_msgs::msg::Float64Stamped` | Publish yaw deviation of self-pose from goal pose [deg]        |
```
Nit: Might be better to insert a space before the unit, and use upper case letters as same as the other tables.

File: common/util/goal_distance_calculator/Readme.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Parameters
+
+### Node Parameters
+
+| Name          | Type   | Default Value | Explanation                 |
+| ------------- | ------ | ------------- | --------------------------- |
+| `update_rate` | double | 10.0          | Timer callback period. [Hz] |
```
Comment (kmiya) [inline common/util/goal_distance_calculator/Readme.md:33]:
```suggestion
| Name          | Type   | Default Value | Explanation                |
| ------------- | ------ | ------------- | -------------------------- |
| `update_rate` | double | 10.0          | Timer callback period [Hz] |
```
Nit: remove extra `.`

File: common/util/goal_distance_calculator/Readme.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+### Core Parameters
+
+| Name      | Type | Default Value | Explanation                                |
+| --------- | ---- | ------------- | ------------------------------------------ |
+| `oneshot` | bool | true          | publish deviations just once or repeatedly |
```
Comment (kmiya) [inline common/util/goal_distance_calculator/Readme.md:39]:
```suggestion
| `oneshot` | bool | true          | Publish deviations just once or repeatedly |
```
Nit: Use upper case letter as same as the other tables.

**Comment by kmiya** [comment]
It has already merged, but just commented.

**Pull request merged**